src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3518,17 +3517,17 @@ private int GetOrCreatePlaceholderSlot(object identifier, TypeWithAnnotations ty
         /// </summary>
         internal static TypeWithAnnotations BestTypeForLambdaReturns(
             ArrayBuilder<(BoundExpression, TypeWithAnnotations)> returns,
-            CSharpCompilation compilation,
+            Binder binder,
             BoundNode node,
             Conversions conversions)
         {
-            var walker = new NullableWalker(compilation,
+            var walker = new NullableWalker(binder.Compilation,
```

Comment:
>binder.Compilation, [](start = 44, length = 19)

Can we remove the compilation from NullableWalker constructor (since we always have binder now)?

Reply:
Probably, but I would want to do that in a separate PR. This is already much bigger than I had originally been intending.

---
In reply to: [549493682](https://github.com/dotnet/roslyn/pull/50105#discussion_r549493682) [](ancestors = 549493682)